types/react-native/index.d.ts

```diff
@@ -3491,7 +3491,7 @@ export interface FlatListProperties<ItemT> extends VirtualizedListProperties<Ite
     /**
      * Rendered in between each item, but not at the top or bottom
      */
-    ItemSeparatorComponent?: React.ComponentClass<any> | null
+    ItemSeparatorComponent?: React.ComponentClass<any> | React.ReactElement<any> | (() => React.ReactElement<any>) | null
```

Comment:
Oops, re-reading my last comment on the issue it seems like I suggested the wrong typing and here again, my bad. I meant `React.ComponentType<any> | null`.

`ReactElement` is afaik the output of a `React.ComponentClass` or `React.StatelessComponent`, which is not what the React Native source expects (it does `<ItemSeparatorComponent … />`)